packages/strapi-hook-mongoose/lib/index.js

```diff
@@ -314,9 +314,11 @@ module.exports = function (strapi) {
               switch (verbose) {
                 case 'hasOne': {
                   const ref = details.plugin ? strapi.plugins[details.plugin].models[details.model].globalId : strapi.models[details.model].globalId;
+                  const customKey = details.plugin ? strapi.plugins[details.plugin].models[details.model].attributes['_id'] : strapi.models[details.model].attributes["_id"];
```

Comment:
Why are you not using the `ref` variable which is the referring model? Also, we should base the customKey on the primaryKey.

```json
{
  "connection": "default",
  "primaryKey": "__uuid",
  "attributes": {
    "__uuid": {
       "type": "string"
    },
    "username": {
      "type": "string",
      "minLength": 3,
      "required": true
    }
}
```

So the code could be improved like that:
```js
const keyType = _.get(ref.attributes, `${ref.primaryKey}.type`, false) ? 
  utils(instance).convertType(ref.attributes[ref.primaryKey].type) : 
  instance.Schema.Types.ObjectId;
```

Reply:
I cant just remember why, but just looking at the code, `ref` looks for the `globalId`... while i need which is the field for the `_id`